pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -0,0 +1,281 @@
+			bc, method := tester.SetLatestValue(&firstItem)
+			secondItem := CreateTestStruct(1, tester.CreateAccount)
+			tester.SetLatestValue(&secondItem)
+
+			cr := tester.GetChainReader()
+			actual := &TestStruct{}
+			params := &LatestParams{I: 1}
+
+			Expect(cr.GetLatestValue(ctx, bc, method, params, actual)).To(Succeed())
+			Expect(actual).To(Equal(&firstItem))
+
+			params.I = 2
+			Expect(cr.GetLatestValue(ctx, bc, method, params, actual)).To(Succeed())
+			Expect(actual).To(Equal(&secondItem))
+		})
+	})
+}
+
+// CreateChainReaderWithStrictArgsInterfaceTest is meant to be used by codecs that don't pad
+// They can assure that the right argument size is verified.
+// Padding makes that harder/impossible to verify for come codecs.
+// However, the extra verification is nice to have when possible.
+func CreateChainReaderWithStrictArgsInterfaceTest(tester ChainReaderInterfaceTester) bool {
+	ctx := context.Background()
+	basicTests := CreateChainReaderInterfaceTests(tester)
+	strictTests := Describe("Codec tests with strict args "+tester.Name(), func() {
+		It("Gives an error decoding extra fields on an item", func() {
+			item := CreateTestStruct(0, tester.CreateAccount)
+			req := &EncodeRequest{
+				TestStructs: []TestStruct{item},
+				ExtraField:  true,
+				TestOn:      TestItemType,
+			}
+			resp := tester.EncodeFields(req)
+			codec := tester.GetChainReader()
+			err := codec.Decode(ctx, resp, &item, TestItemType)
+			Expect(err).To(Equal(types.InvalidEncodingError{}))
+		})
+
+		It("Gives an error decoding missing fields on an item", func() {
+			item := CreateTestStruct(0, tester.CreateAccount)
+			req := &EncodeRequest{
+				TestStructs:  []TestStruct{item},
+				MissingField: true,
+				TestOn:       TestItemType,
+			}
+			resp := tester.EncodeFields(req)
+			codec := tester.GetChainReader()
+			Expect(codec.Decode(ctx, resp, &item, TestItemType)).To(Equal(types.InvalidEncodingError{}))
+		})
+
+		It("Gives an error decoding extra fields on a slice", func() {
+			items := []TestStruct{CreateTestStruct(0, tester.CreateAccount)}
+			req := &EncodeRequest{
+				TestStructs: items,
+				ExtraField:  true,
+				TestOn:      TestItemSliceType,
+			}
+			resp := tester.EncodeFields(req)
+			codec := tester.GetChainReader()
+			err := codec.Decode(ctx, resp, &items, TestItemSliceType)
+			Expect(err).To(Equal(types.InvalidEncodingError{}))
+		})
+
+		It("Gives an error decoding missing fields on an slice", func() {
+			items := []TestStruct{CreateTestStruct(0, tester.CreateAccount)}
+			req := &EncodeRequest{
+				TestStructs:  items,
+				MissingField: true,
+				TestOn:       TestItemSliceType,
+			}
+			resp := tester.EncodeFields(req)
+			codec := tester.GetChainReader()
+			err := codec.Decode(ctx, resp, &items, TestItemSliceType)
+			Expect(err).To(Equal(types.InvalidEncodingError{}))
+		})
+	})
+	return basicTests && strictTests
+}
+
+func Must[T any](t T, err error) T {
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
How does a failure from this point manifest? Does this framework panic internally?

Reply:
I think it panics internally, I'm not 100% sure.  Never needed to dig into it.

When initializing it is passed t, it may use something from there.   That code is in a different file.   I'll remove the library anyway.

`RunSpecs(t, "loop internal interface tests")`